### PR TITLE
Update to pass callback for nodejs4.3 runtime

### LINKED ISSUE #70: Support new Node v4.3 API when running locally
Lambda now supports Node.js v4.3 and it has changed the API, most notably `done()`, `succeed()`, `fail()` are deprecated on the `context` object in favor of a `callback` object:

```
function handler (event, context, callback) {
...
}
```

`node-lambda` should support either API based on the `AWS_RUNTIME` value.

### PR DESCRIPTION
**Pull request by briandonahue**
Fixes #70 - Passing along runtime to `run` method" and using a callback to handle success/fail/done scenarios.  Added deprecation warning for `context.[succeed,fail,done]()` calls.

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -45,25 +45,53 @@ Lambda.prototype.run = function (program) {
...
+      process.exit(0);
+    }
+  };
+
   var context = {
+    isNode43: runtime === "nodejs4.3",
     succeed: function (result) {
-      console.log('succeed: ' + JSON.stringify(result));
-      process.exit(0);
+      if(isNode43) console.log('context.succeed() is deprecated with Node.js 4.3 runtime');
```
Comment (DeviaVir) [inline lib/main.js]: Could you rewrite this as `if (isNode43) {` and the corresponding `}` on a newline?

Comment (DeviaVir) [inline lib/main.js]: It's also not entirely clear; it is deprecated, but can you still use it? Should we kill succeed/fail/done for anything `isNode43`?

**Comment by DeviaVir** [comment]
Would you mind updating https://github.com/motdotla/node-lambda/blob/master/README.md#run with this extra parameter as well? And perhaps make a note of what runtimes are currently available for both?

Greatly appreciating this work so far!

File: lib/main.js (hunk 1 above)
Comment (briandonahue) [inline lib/main.js]: It is supported for backward compatibility: http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-using-old-runtime.html#transition-to-new-nodejs-runtime

**Comment by briandonahue** [comment]
OK, I also changed the default runtime to `nodejs4.3` since that is now the recommended version, and I added some detail to the README.  Not sure if we want to put a note in the README somewhere pointing to the docs re: the version differences?

**Comment by DeviaVir** [comment]
@briandonahue That would be great actually, could you add a new paragraph below https://github.com/motdotla/node-lambda#custom-environment-variables ?
Will merge this when that's in!

**Comment by DeviaVir** [comment]
Thanks @briandonahue!

**Comment by briandonahue** [comment]
Sure thing - thanks to all the contributors, node-lambda has made playing with AWS Lambda stuff much easier!

**Pull request merged**